scripts/global/player_inventory.gd

```diff
@@ -0,0 +1,142 @@
+extends Node
+
+signal active_item_updated
+signal inventory_active_item_updated
+
+# Constants defining the number of inventory and hotbar slots
+const NUM_INVENTORY_SLOTS = 20
```

Comment:
Please define types. It's annoying but will save you from 50% of bugs you'll encounter in normal work.